decomp2dbg/clients/gdb/gdb_client.py

```
@@ -148,29 +153,50 @@ def update_function_data(self, addr):
                 except Exception:
                     continue
 
-        for offset, stack_var in stack_vars.items():
-            offset = abs(int(offset, 0))
+        for stack_var in stack_vars:
+            var_name = stack_var['name']
             type_str = self._clean_type_str(stack_var['type'])
-            lvar_bptr = self._find_local_var_base_ptr()
-            if lvar_bptr == "$rbp":
-                offset -= 8
-            elif lvar_bptr == "$ebp":
-                offset -= 4
+            # Have to use .get() because of ghidra
+            from_sp_str: None | str = stack_var.get("from_sp")
+            from_frame_str: None | str = stack_var.get("from_frame")
+
+            # We prefer from sp, because sp always exists while it may be
+            # hard/unstable/impossible to find the frame
+            # We don't account for architectures where the stack goes in the
+            # different direction.
+            if from_sp_str is not None:
+                from_sp: int = int(from_sp_str, 0)
+                try:
+                    gdb.execute(f"set ${var_name} = ({type_str}*) ($sp + {from_sp})")
+                    type_unknown = False
+                except Exception:
+                    type_unknown = True
 
-            expr = f"""({type_str}*) ({lvar_bptr} - {offset})"""
-            var_name = stack_var['name']
+                if type_unknown:
+                    try:
+                        gdb.execute(f"set ${var_name} = ($sp + {from_sp})")
+                    except Exception:
+                        continue
+            else:
+                if from_frame_str is None:
+                    raise ValueError("Both from_frame and from_sp are None, shouldn't be possible.")
```

Although it usually is good to do this, could you avoid a hard error here? Our goal as a plugin extension is not to error almost ever, so at least some things still work.

---

i generally also have this sentiment, but in this case no matter what user does or the decompiler returns it won't happen, so it must be a programmer error if it happens, in which case i think it makes sense to raise it to the user so they can file a bug report (rather than variables silently not being added which would go unnoticed for long)

but oke, its not a problem to change it